perception/object_recognition/tracking/multi_object_tracker/src/multi_object_tracker_core.cpp

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2018 Autoware Foundation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * v1.0 Yukihiro Saito
+ */
+
+#include "multi_object_tracker/multi_object_tracker_core.hpp"
+#include <rclcpp_components/register_node_macro.hpp>
+
+#include <tf2_ros/create_timer_ros.h>
+#include <tf2_ros/create_timer_interface.h>
+#define EIGEN_MPL2_ONLY
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+#include <string>
+
+MultiObjectTracker::MultiObjectTracker(const rclcpp::NodeOptions & node_options)
+: rclcpp::Node("multi_object_tracker", node_options)
+{
+  // Create publishers and subscribers
+  dynamic_object_sub_ =
+    create_subscription<autoware_perception_msgs::msg::DynamicObjectWithFeatureArray>(
+      "input", rclcpp::QoS{1},
+      std::bind(&MultiObjectTracker::measurementCallback, this, std::placeholders::_1));
+  dynamic_object_pub_ =
+    create_publisher<autoware_perception_msgs::msg::DynamicObjectArray>("output", rclcpp::QoS{1});
+
+  // Parameters
+  double publish_rate = declare_parameter<double>("publish_rate", 30.0);
+  world_frame_id_ = declare_parameter<std::string>("world_frame_id", std::string("world"));
```

Comment:
I think the two explicit `std::string` are not needed (but it's fine to leave it in if you prefer).

Reply:
Good point, changed in 0febc64